### ISSUE TITLE
don't create hints in aria-hidden or inert elements

### ISSUE THREAD

**Issue by codl** [qutebrowser/qutebrowser#3869]
Web applications ofter use full-screen modals and overlay a backdrop over the rest of the application. They use [`aria-hidden="true"`][aria-hidden] or [`inert="true"`][inert] on the rest of the page to signal to screen readers and other assistive technologies that it is currently inaccessible.

Qutebrowser should respect those attributes and not offer any hints for elements and children of elements with `aria-hidden="true"` or `inert="true"`

![an example of a web app with a modal open, and dozens of hints being displayed for buttons that are underneath the modal and should not be accessible](https://user-images.githubusercontent.com/315139/39330496-1ede1072-49a2-11e8-9442-db3eedf07162.png)

[aria-hidden]: https://www.w3.org/WAI/PF/aria/states_and_properties#aria-hidden
[inert]: https://github.com/WICG/inert/blob/master/README.md

**Comment by jgkamat** [comment]
Something like 


```diff
diff --git a/qutebrowser/javascript/webelem.js b/qutebrowser/javascript/webelem.js
index f7ab0f636..071eb347e 100644
--- a/qutebrowser/javascript/webelem.js
+++ b/qutebrowser/javascript/webelem.js
@@ -219,7 +219,8 @@ window._qutebrowser.webelem = (function() {
         const out = [];
 
         for (let i = 0; i < elems.length; ++i) {
-            if (!only_visible || is_visible(elems[i])) {
+            if (!only_visible || is_visible(elems[i])
+                    && !elems[i].matches("[aria-hidden=true] *")) {
                 out.push(serialize_elem(elems[i]));
             }
         }
```

seems to work at least on twitter. This will probably conflict with #3806 in some way, so this shold probably wait for that to push through. I wonder if we can get away with having a hint blacklist selector which is applied always, to reduce complexity.
